hack/components/common_functions.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -xe
+
+function fetch_component() {
```

Comment:
- The order of the commits seem wrong, as this change comes after the one that calls it.
- Lets try to avoid "common", "utils" and friends, they just point us to the junkyard of all operations we do not know to name. How about calling this one "component"? BTW, what components are these?
- When using `source`, we loose context of where this is coming from. I saw Daniel B. using <filename>::<funcname>, e.g. `component::fetch`.

Reply:
- I rechecked the order of the commits, can you explain what you find ammis with it?
- I agree, I'll change it into 2 scripts: "git_utils.sh" and "yaml_utils.sh"
- I agree, will change the functions names accordingly